crates/sequencer/src/sequencer.rs

```diff
@@ -591,94 +597,153 @@ where
-                                l2_end,
-                                commitment.merkle_root.to_vec(),
-                                CommitmentStatus::Mempool,
-                            )
-                            .await
-                            .map_err(|_| {
-                                anyhow!("Sequencer: Failed to insert sequencer commitment")
-                            })?;
-                    }
-                    Err(e) => {
-                        warn!("Failed to connect to postgres: {:?}", e);
+    #[instrument(level = "trace", skip(self), err, ret)]
+    pub async fn resubmit_pending_commitments(&mut self) -> anyhow::Result<()> {
+        let pending_commitments = self.pending_commitments_l2_range.lock().await.clone();
+        for (l2_start, l2_end) in pending_commitments {
+            let commitment_info = commitment_controller::CommitmentInfo {
+                l2_height_range: l2_start..=l2_end,
+            };
+            self.submit_commitment(commitment_info, true).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn submit_commitment(
+        &mut self,
+        commitment_info: commitment_controller::CommitmentInfo,
+        resubmit: bool,
+    ) -> anyhow::Result<()> {
+        let l2_range_to_submit = commitment_info.l2_height_range.clone();
+
+        // calculate exclusive range end
+        let range_end = BatchNumber(l2_range_to_submit.end().0 + 1); // cannnot add u64 to BatchNumber directly
+
+        let soft_confirmation_hashes = self
+            .ledger_db
+            .get_soft_batch_range(&(*l2_range_to_submit.start()..range_end))?
+            .iter()
+            .map(|sb| sb.hash)
+            .collect::<Vec<[u8; 32]>>();
+
+        let commitment = commitment_controller::get_commitment(
+            commitment_info.clone(),
+            soft_confirmation_hashes,
+        )?;
+
+        debug!("Sequencer: submitting commitment: {:?}", commitment);
+
+        let blob = borsh::to_vec(&DaData::SequencerCommitment(commitment.clone()))
+            .map_err(|e| anyhow!(e))?;
+        let (notify, rx) = oneshot_channel();
+        let request = BlobWithNotifier { blob, notify };
+        self.da_service
+            .get_send_transaction_queue()
+            .send(request)
+            .map_err(|_| anyhow!("Bitcoin service already stopped!"))?;
+
+        info!(
+            "Sent commitment to DA queue. L2 range: #{}-{}",
+            commitment.l2_start_block_number, commitment.l2_end_block_number
+        );
+
+        let ledger_db = self.ledger_db.clone();
+        let db_config = self.config.db_config.clone();
+        let pending_commitments_l2_range = self.pending_commitments_l2_range.clone();
+        let handle_da_response = async move {
+            let result: anyhow::Result<()> = async move {
+                let l2_start = *commitment_info.l2_height_range.start();
+                let l2_end = *commitment_info.l2_height_range.end();
+
+                let tx_id = rx
```

Comment:
This is a oneshot channel which is going to be closed when a single signal is received.... cannot be reused.